src/Components/ServiceScene/Breakdowns/FilterByPatternsButton.tsx

```diff
@@ -1,53 +1,52 @@
 import React from 'react';
 
-import { SceneObjectState, SceneObjectBase, SceneComponentProps, sceneGraph } from '@grafana/scenes';
+import { SceneComponentProps, sceneGraph, SceneObjectBase, SceneObjectState } from '@grafana/scenes';
 import { Button } from '@grafana/ui';
 import { IndexScene } from '../../IndexScene/IndexScene';
-import { USER_EVENTS_ACTIONS, USER_EVENTS_PAGES, reportAppInteraction } from 'services/analytics';
+import { reportAppInteraction, USER_EVENTS_ACTIONS, USER_EVENTS_PAGES } from 'services/analytics';
 
 export interface FilterByPatternsButtonState extends SceneObjectState {
   pattern: string;
   type: 'exclude' | 'include';
 }
 
-export class FilterByPatternsButton extends SceneObjectBase<FilterByPatternsButtonState> {
-  public onClick = () => {
-    const logExploration = sceneGraph.getAncestor(this, IndexScene);
+export interface FilterByPatternsState extends FilterByPatternsButtonState {
+  indexScene: IndexScene;
+}
 
-    if (!logExploration) {
-      return;
-    }
+// @todo refactor
```

Comment:
What in particular should be refactored? Mainly wondering whether to keep this comment or not, as it's also applicable in many other places.

Reply:
Fixed, I had wanted to get rid of the class that was only used by the other views, which we removed, so it's been cleaned up now!